src/track.cpp

```diff
@@ -62,8 +62,11 @@ bool Track::send(message_variant data) {
 
 	auto message = make_message(std::move(data));
 
-	if (mRtcpHandler) {
-		message = mRtcpHandler->outgoing(message);
+    std::shared_lock lock(mRtcpHandlerMutex);
+    if (mRtcpHandler) {
+        auto copy = mRtcpHandler;
+        lock.unlock();
```

Comment:
This whole repeated block could be made shorter with
'''
if (auto copy = getRtcpHandler()) copy->...
'''